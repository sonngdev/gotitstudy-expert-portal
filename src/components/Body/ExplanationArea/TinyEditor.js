import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { weakRandomString, useTinyMceEditor } from '../../../utils';

/**
 * Custom tinymce React component:
 * https://www.tiny.cloud/blog/how-to-integrate-react-with-tinymce/
 */
export function TinyEditor({ id, value, onChange }) {
  const [input, setInput] = useState(value);
  const bindAndCall = (val) => {
    setInput(val);
    onChange(val);
  };

  useTinyMceEditor(id, bindAndCall);

  return (
    <textarea id={id} value={input} onChange={(e) => bindAndCall(e.target.value)} />
  );
}

TinyEditor.propTypes = {
  id: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
};

TinyEditor.defaultProps = {
  id: `tinymce-${weakRandomString(5)}`,
  value: '',
  onChange: () => {},
};
