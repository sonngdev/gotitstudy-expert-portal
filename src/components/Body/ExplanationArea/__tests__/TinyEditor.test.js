import React from 'react';
import { shallow } from 'enzyme';
import { TinyEditor } from '../TinyEditor';

describe('<TinyEditor />', () => {
  let wrapper;
  let onChange;

  beforeEach(() => {
    onChange = jest.fn();

    wrapper = shallow(
      <TinyEditor
        id="tiny-editor"
        value="foo"
        onChange={onChange}
      />,
    );
  });

  it('renders', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('has given initial value', () => {
    expect(wrapper.find('textarea').prop('value')).toBe('foo');
  });

  it('binds input', () => {
    wrapper.find('textarea').simulate('change', { target: { value: 'bar' } });
    expect(wrapper.find('textarea').prop('value')).toBe('bar');
  });

  it('calls onChange each time input changes', () => {
    wrapper.find('textarea').simulate('change', { target: { value: 'baz' } });
    expect(onChange).toHaveBeenCalledWith('baz');
  });
});
