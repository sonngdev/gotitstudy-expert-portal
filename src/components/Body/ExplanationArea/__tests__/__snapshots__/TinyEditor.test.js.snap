// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TinyEditor /> renders 1`] = `"<textarea id=\\"tiny-editor\\" value=\\"foo\\" onChange={[Function: onChange]} />"`;
